Refactoring/gcp/cloudresourcemanager

Remi05 opened the pull request:
Migrated GCP's Cloud Resource Manager service to the new architecture.

Note that `gcp/resources/projects.py` and `gcp/facade/base.py` are the same as in the [Stackdriver logging PR](https://github.com/nccgroup/ScoutSuite/pull/229) and the [IAM PR](https://github.com/nccgroup/ScoutSuite/pull/230); they are simply shared by all services and will appear in all GCP refactoring PRs until one of the services is merged.

j4v commented:
:clap: :clap: :clap:

Aboisier commented on ScoutSuite/providers/gcp/resources/cloudresourcemanager/bindings.py:
```diff
@@ -0,0 +1,44 @@
+        }
+        
+        for member in raw_binding['members']:
+            member_type, entity = member.split(':')[:2]
```
Maybe add a comment explaining this line, it's a bit esoteric if you don't know what the raw object looks like.

misg commented on ScoutSuite/providers/gcp/facade/cloudresourcemanager.py:
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
+
```
```suggestion
from ScoutSuite.providers.utils import run_concurrently
```

misg commented on ScoutSuite/providers/gcp/facade/cloudresourcemanager.py:
```diff
@@ -0,0 +1,14 @@
+
+    # TODO: Make truly async
+    async def get_bindings(self, project_id):
+        return self._resourcemanager_client.projects().getIamPolicy(resource=project_id).execute()
```
```suggestion
        return await run_concurrently(
                lambda: self._resourcemanager_client.projects().getIamPolicy(resource=project_id).execute()
        )
```

Remi05 commented:
Added left menu for projects and fixed UI.

Before : 

![cloudresourcemanager_before](https://user-images.githubusercontent.com/12770956/55372448-c913eb00-54cf-11e9-8f61-8c9feef45337.png)

After : 

![cloudresourcemanager_after](https://user-images.githubusercontent.com/12770956/55372452-cd400880-54cf-11e9-8b0d-96e40897dc4c.png)

Aboisier commented on ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py:
```diff
@@ -0,0 +1,13 @@
+    def __init__(self, gcp_facade: GCPFacade):
+        super(CloudResourceManager, self).__init__(gcp_facade)
+
+        
```
```suggestion
```

Aboisier commented on ScoutSuite/providers/gcp/resources/cloudresourcemanager/service.py:
```diff
@@ -0,0 +1,13 @@
+    ]
+
+    def __init__(self, gcp_facade: GCPFacade):
+        super(CloudResourceManager, self).__init__(gcp_facade)
```
Is this constructor necessary? It's the same as the base.

Remi05 replied:
Yeah, after some additional refactoring it became obsolete. Good catch!
